.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+  pull_request:
+    branches:
+      - master
+      - develop
+
+#env:
+#  CLAZY_CHECKS: "level1"
+
+jobs:
+  build:
+    if: "!contains(github.event.head_commit.message, 'ci skip')"
+    strategy:
+      matrix:
+        compiler:
+          - gcc
+          - clang
+        container:
+          #- suse-qt512
+          - suse-qt515
+        pam:
+          - ON
+          - OFF
+    runs-on: ubuntu-latest
+    container:
+      image: liridev/${{ matrix.container }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Dependencies
+        run: |
+          set -x
+          zypper install python-docutils
+      - name: Build
+        run: |
+          set -x
+          mkdir -p build
+          cd build
+          if [ "${{ matrix.compiler }}" == "clang" ]; then
+            export CC=clang
+            export CXX=clang++
+            #export CXX=clazy
+          fi
+          cmake .. \
+            -DBUILD_MAN_PAGES:BOOL=ON \
+            -DENABLE_PAM:BOOL=${{ matrix.pam }} \
+            -DLOGIN_DEFS_PATH:path="/usr/etc/login.defs"
+          make -j $(getconf _NPROCESSORS_ONLN)
+          sudo make install
+      - name: Notification for push events
+        uses: rectalogic/notify-irc@v1
```

Comment:
References to third-party actions should use the full sha1.

Reply:
that's cumbersome, why not tags?

Reply:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses

it's not mandatory to use a full sha1

```
We strongly recommend that you include the version of the action you are using by specifying a Git ref, SHA, or Docker tag number. If you don't specify a version, it could break your workflows or cause unexpected behavior when the action owner publishes an update.
```

tags are fine:

```
    Using the commit SHA of a released action version is the safest for stability and security.
    Using the specific major action version allows you to receive critical fixes and security patches while still maintaining compatibility. It also assures that your workflow should still work.
    Using the default branch of an action may be convenient, but if someone releases a new major version with a breaking change, your workflow could break.
```

Reply:
Security reasons: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions

Reply:
ok thanks

i had to remove the irc notification though because it doesn't work, i wll search one that works later